src/components/ThemeProvider/tests/ThemeProvider.test.tsx

```diff
@@ -164,4 +165,221 @@ describe('<ThemeProvider />', () => {
+      );
+
+      const {style} = wrapper
+        .find('div')
+        .last()
+        .props();
+      expect(style).toStrictEqual(
+        expect.objectContaining({
+          '--p-critical-surface-subdued':
+            'hsla(58, 100%, 7.000000000000001%, 1)',
+        }),
+      );
+      expect(style).not.toStrictEqual(
+        expect.objectContaining({
+          '--p-critical-surface-subdued': 'hsla(57, 100%, 89%, 1)',
+        }),
+      );
+    });
+
+    it('overrides mode from parent <ThemeProvider> when provided a mode', () => {
+      const wrapper = mountWithGlobalTheming(
+        <ThemeProvider theme={{mode: 'dark'}}>
+          <ThemeProvider
+            theme={{
+              UNSTABLE_colors: {critical: '#FFFEEE'},
+              mode: 'light',
+            }}
+          >
+            <p>Hello</p>
+          </ThemeProvider>
+        </ThemeProvider>,
+        true,
+      );
+
+      const {style} = wrapper
+        .find('div')
+        .last()
+        .props();
+      expect(style).toStrictEqual(
+        expect.objectContaining({
+          '--p-critical-surface-subdued': 'hsla(57, 100%, 89%, 1)',
+        }),
+      );
+      expect(style).not.toStrictEqual(
+        expect.objectContaining({
+          '--p-critical-surface-subdued':
+            'hsla(58, 100%, 7.000000000000001%, 1)',
+        }),
+      );
+    });
+
+    it('inherits colors from parent <ThemeProvider> when their modes differ', () => {
+      const wrapper = mountWithGlobalTheming(
+        <ThemeProvider theme={{mode: 'dark'}}>
+          <ThemeProvider
+            theme={{
+              mode: 'light',
+            }}
+          >
+            <p>Hello</p>
+          </ThemeProvider>
+        </ThemeProvider>,
+        true,
+      );
+
+      const {style} = wrapper
+        .find('div')
+        .last()
+        .props();
+      expect(style).toStrictEqual(
+        expect.objectContaining({
+          '--p-surface-background': 'hsla(0, 0%, 98%, 1)',
+        }),
+      );
+    });
+  });
+
+  it('overrides inherited colors from parent <ThemeProvider> with provided colors when their modes differ', () => {
+    const wrapper = mountWithGlobalTheming(
+      <ThemeProvider
+        theme={{
+          mode: 'dark',
+          UNSTABLE_colors: {critical: '#000000'},
+        }}
+      >
+        <ThemeProvider
+          theme={{
+            mode: 'light',
+            UNSTABLE_colors: {critical: '#FFFEEE'},
+          }}
+        >
+          <p>Hello</p>
+        </ThemeProvider>
+      </ThemeProvider>,
+      true,
+    );
+
+    const {style} = wrapper
+      .find('div')
+      .last()
+      .props();
+    expect(style).toStrictEqual(
+      expect.objectContaining({
+        '--p-critical-surface': 'hsla(57, 100%, 93%, 1)',
+      }),
+    );
+    expect(style).not.toStrictEqual(
+      expect.objectContaining({
+        '--p-critical-surface': 'hsla(0, 0%, 98%, 1)',
+      }),
+    );
+  });
 });
+
+function mountWithGlobalTheming(
```

Comment:
Using mount's wrappingComponent options might be a little cleaner - and it also means that the node you get back  from calling `mount(someNode)` is  the node you passed in as the argument instead of the wrapping context provider. Perhaps something like this (inspired by our implementation of mountWithAppProvider)

```ts
export function mountWithGlobalTheming<P>(
  node: React.ReactElement<P>,
  unstableGlobalTheming: boolean,
) {
  return mount<P>(node, {
    wrappingComponent: FeaturesContext.Provider,
    wrappingComponentProps: {value: {unstableGlobalTheming}},
  });
}
```

Reply:
Alternativly using a custom mount from react-testing would be preferable. Happy to pair with ya for these test updates next week.

Reply:
Pairing next week on the test updates would be great! I will send you an invite